core/error.rs

```diff
@@ -1106,8 +1106,38 @@ pub(crate) fn has_call_site<'s, 'i>(
 
 const DATA_URL_ABBREV_THRESHOLD: usize = 150;
 
-pub fn format_file_name(file_name: &str) -> String {
-  abbrev_file_name(file_name).unwrap_or_else(|| {
+fn to_percent_decoded_str(s: &str) -> String {
+  match percent_encoding::percent_decode_str(s).decode_utf8() {
+    Ok(s) => s.to_string(),
+    // when failed to decode, return the original string
+    Err(_) => s.to_string(),
+  }
+}
+
+pub fn relative_specifier(from: &Url, to: &Url) -> Option<String> {
+  let is_dir = to.path().ends_with('/');
+
+  if is_dir && from == to {
+    return Some("./".to_string());
+  }
+
+  // workaround for url crate not adding a trailing slash for a directory
+  // it seems to be fixed once a version greater than 2.2.2 is released
```

Comment:
Is this pending a release from a `url` crate then?

Reply:
Ah copied it from the CLI, I'm pretty sure it's already on a newer version. I'll remove that stuff